apps/chef_objects/src/chef_regex.erl

```diff
@@ -100,6 +100,10 @@ regex_for(cookbook_name) ->
 regex_for(environment_name) ->
     generate_regex_msg_tuple(?ANCHOR_REGEX(?NAME_REGEX),
                              <<"Malformed environment name. Must only contain A-Z, a-z, 0-9, _ or -">>);
+regex_for(key_name) ->
+    % This might be the same as nodename -- nodename seems to allow ':' as well
+    generate_regex_msg_tuple(?ANCHOR_REGEX(?NAME_REGEX),
```

Comment:
I believe allowing ':' is required given our current tooling.  The key_name, if one is not supplied by the user, will be the fingerprint of the key with ':' separating the hex digits:

https://github.com/chef/opscode-omnibus/blob/master/files/private-chef-ctl-commands/key-control.rb#L129